### DISCUSSION TITLE
Difference between XSharedPreferences and XRemotePreferences

### DISCUSSION THREAD

**Discussion by Ninlives** [LSPosed/LSPosed#1182] [Q&A]
So I noticed that there are two implementations of SharedPreference in the repo, what's the difference, advantage and disadvantage of each implmentation? Also, what's the recommended way to used XRemotePreferences in module since it is not provided by the original xposed api?

**Comment by Ninlives** [comment]
Apologize for the duplicated question, how can I delete this one?

**Comment by yujincheng08** [comment]
`XRemotePreferences` is binder-based.

**Reply by Ninlives** [reply]
Sorry I'm not familiar about the android internal, so what's the advantage of binder-based implementations? It must have a good reason to provide an alternative.
